src/core_plugins/kibana/public/dashboard/directives/grid.js

```diff
@@ -119,91 +138,53 @@ app.directive('dashboardGrid', function ($compile, Notifier) {
-        delete panel.$scope;
-      }
-
       // tell gridster to remove the panel, and cleanup our metadata
       function removePanel(panel, silent) {
         // remove from grister 'silently' (don't reorganize after)
         gridster.remove_widget(panel.$el, silent);
-
-        // destroy the scope
-        panel.$scope.$destroy();
-
         panel.$el.removeData('panel');
-        panel.$el.removeData('$scope');
       }
 
       // tell gridster to add the panel, and create additional meatadata like $scope
       function addPanel(panel) {
-        _.defaults(panel, {
-          size_x: 3,
-          size_y: 2
-        });
-
-        // ignore panels that don't have vis id's
-        if (!panel.id) {
-          // In the interest of backwards compat
-          if (panel.visId) {
-            panel.id = panel.visId;
-            panel.type = 'visualization';
-            delete panel.visId;
-          } else {
-            throw new Error('missing object id on panel');
-          }
-        }
-
-        panel.$scope = $scope.$new();
-        panel.$scope.panel = panel;
-        panel.$scope.parentUiState = $scope.uiState;
+        PanelUtils.initializeDefaults(panel);
 
-        panel.$el = $compile('<li><dashboard-panel></li>')(panel.$scope);
+        const panelHtml = `
+            <li>
+                <dashboard-panel remove="removePanelFromState(${panel.panelId})"
+                                 panel="getPanelByPanelId(${panel.panelId})"
+                                 isFullScreenMode="!chrome.getVisible()"
+                                 parent-ui-state="uiState">
+            </li>`;
+        panel.$el = $compile(panelHtml)($scope);
 
         // tell gridster to use the widget
         gridster.add_widget(panel.$el, panel.size_x, panel.size_y, panel.col, panel.row);
 
-        // update size/col/etc.
-        refreshPanelStats(panel);
+        // Gridster may change the position of the widget when adding it, make sure the panel
+        // contains the latest info.
+        PanelUtils.refreshSizeAndPosition(panel);
 
-        // stash the panel and it's scope in the element's data
+        // stash the panel in the element's data
         panel.$el.data('panel', panel);
-        panel.$el.data('$scope', panel.$scope);
-      }
-
-      // ensure that the panel object has the latest size/pos info
-      function refreshPanelStats(panel) {
-        const data = panel.$el.coords().grid;
-        panel.size_x = data.size_x;
-        panel.size_y = data.size_y;
-        panel.col = data.col;
-        panel.row = data.row;
       }
 
       // when gridster tell us it made a change, update each of the panel objects
       function readGridsterChangeHandler(e, ui, $widget) {
         // ensure that our panel objects keep their size in sync
         gridster.$widgets.each(function (i, el) {
           const panel = getPanelFor(el);
-          refreshPanelStats(panel);
-          panel.$scope.$broadcast('resize');
```

Comment:
Removing it was intentional since it seemed like no one was listening to it.  I couldn't find any `$scope.$on('resize'`, though I do see `$window.on('resize', safeLayout);` but I'm pretty sure that doesn't listen to the broadcast events... I'll have to double check this to make sure though...